misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/BallerinaStackFrame.java

```diff
@@ -128,13 +133,37 @@ private static String getStackFrameName(StackFrameProxyImpl stackFrame) {
                 return stackFrame.visibleVariableByName(STRAND_VAR_NAME) != null ? frameName :
                         FRAME_TYPE_START + FRAME_SEPARATOR + frameName;
             } else {
-                return stackFrame.location().method().name();
+                return getFilteredStackFrame(stackFrame);
             }
         } catch (Exception e) {
             return FRAME_TYPE_ANONYMOUS;
         }
     }
 
+    /**
+     * Derives filtered ballerina stack frame name from the given java stack frame instance.
+     *
+     * @param stackFrame JDI stack frame instance
+     * @return filtered ballerina stack frame name
+     */
+    private static String getFilteredStackFrame(StackFrameProxyImpl stackFrame) throws JdiProxyException {
+        Value value = stackFrame.getValue(stackFrame.visibleVariables().get(0));
```

Comment:
We cannot guarantee that the 0th element will always be the service variable. Lets modify the logic to check for `self` variable.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31279/commits/8b8ba6a04de50d8d9a98f0622372775b3c2b9262